internal/impl/splunk/input.go

```diff
@@ -0,0 +1,181 @@
+
+func inputSpec() *service.ConfigSpec {
+	return service.NewConfigSpec().
+		// Stable().
+		Version("4.29.1").
+		Categories("Services").
+		Summary(`Publishes messages to Splunk.`).
+		Description(service.OutputPerformanceDocs(true, true)).
+		Fields(
+			service.NewStringField(siFieldURL).Description("Full HTTP Search API endpoint URL.").Example("https://foobar.splunkcloud.com/services/search/v2/jobs/export"),
+			service.NewStringField(siFieldUser).Description("Splunk account user."),
+			service.NewStringField(siFieldPassword).Description("Splunk account password.").Secret(),
+			service.NewStringField(siFieldQuery).Description("Splunk search query."),
+			service.NewBoolField(siFieldSkipCertVerify).Description("Whether to skip server side certificate verification.").Advanced().Default(false),
+			service.NewAutoRetryNacksToggleField(),
+		)
+}
+
+func init() {
+	err := service.RegisterInput("splunk", inputSpec(),
+		func(conf *service.ParsedConfig, mgr *service.Resources) (service.Input, error) {
+			i, err := inputFromParsed(conf)
+			if err != nil {
+				return nil, err
+			}
+			return service.AutoRetryNacksToggled(conf, i)
+		})
+	if err != nil {
+		panic(err)
+	}
+}
+
+type input struct {
+	url      string
+	user     string
+	password string
+	query    string
+
+	client    http.Client
+	reader    *bufio.Reader
+	clientMut sync.Mutex
+	shutSig   *shutdown.Signaller
+}
+
+func inputFromParsed(pConf *service.ParsedConfig) (o *input, err error) {
+	o = &input{
+		shutSig: shutdown.NewSignaller(),
+	}
+
+	if o.url, err = pConf.FieldString(siFieldURL); err != nil {
+		return
+	}
+
+	if o.user, err = pConf.FieldString(siFieldUser); err != nil {
+		return
+	}
+
+	if o.password, err = pConf.FieldString(siFieldPassword); err != nil {
+		return
+	}
+
+	if o.query, err = pConf.FieldString(siFieldQuery); err != nil {
+		return
+	}
+
+	var skipCertVerify bool
+	if skipCertVerify, err = pConf.FieldBool(siFieldSkipCertVerify); err != nil {
+		return
+	}
+
+	tr := http.DefaultTransport.(*http.Transport).Clone()
+	tr.TLSClientConfig.InsecureSkipVerify = skipCertVerify
+	o.client = http.Client{Transport: tr}
+
+	return
+}
+
+//------------------------------------------------------------------------------
+
+func (i *input) Connect(ctx context.Context) error {
+	i.clientMut.Lock()
+	defer i.clientMut.Unlock()
+
+	payload := make(url.Values)
+	payload.Set("search", "search "+i.query)
+	payload.Set("output_mode", "json")
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, i.url, strings.NewReader(payload.Encode()))
+	if err != nil {
+		return fmt.Errorf("failed to construct http request: %s", err)
+	}
+	req.SetBasicAuth(i.user, i.password)
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+
+	time.Sleep(5 * time.Second)
+
+	resp, err := i.client.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to execute http request: %s", err)
+	}
+
+	// TODO: HTTP ERROR HANDLING
```

Comment:
More granular error handling is a nice-to-have but what we have here seems reasonable to me as a starting point.

Reply:
I'll run a few more tests and see if it's worth adding some checks to aid users.

Reply:
I ended up checking the status as well. OTEL is doing some extra handling [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/2a3c4933fb656aa38618935f62d197091f998107/exporter/splunkhecexporter/hec_worker.go#L65-L72) for status 429 or 502, but I'm not sure if that's relevant for Connect so I didn't replicate it for now.